sdk/highlight-grafana-datasource/pkg/plugin/datasource.go

```diff
@@ -0,0 +1,252 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/hasura/go-graphql-client"
+	"github.com/samber/lo"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Make sure Datasource implements required interfaces. This is important to do
+// since otherwise we will only get a not implemented error response from plugin in
+// runtime. In this example datasource instance implements backend.QueryDataHandler,
+// backend.CheckHealthHandler interfaces. Plugin should not implement all these
+// interfaces - only those which are required for a particular task.
+var (
+	_ backend.QueryDataHandler      = (*Datasource)(nil)
+	_ backend.CheckHealthHandler    = (*Datasource)(nil)
+	_ instancemgmt.InstanceDisposer = (*Datasource)(nil)
+)
+
+// NewDatasource creates a new datasource instance.
+func NewDatasource(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	opts, err := settings.HTTPClientOptions(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+
+	opts.ForwardHTTPHeaders = true
+
+	var dataSourceSettings DataSourceSettings
+	err = json.Unmarshal(settings.JSONData, &dataSourceSettings)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSecret := settings.DecryptedSecureJSONData["clientSecret"]
+
+	config := clientcredentials.Config{
+		ClientID:     dataSourceSettings.ClientId,
+		ClientSecret: clientSecret,
+		TokenURL:     "https://pri.highlight.io/oauth/token",
```

Comment:
perhaps the hostname could be a setting too, to support on-prem highlight deploys?